larana/ParticleIdentification/MVAAlg.cxx

```diff
@@ -303,7 +303,8 @@ void mvapid::MVAAlg::PrepareEvent(const art::Event& evt,
 
   for (unsigned int iTrack = 0; iTrack < fTracks.size(); ++iTrack) {
     const art::Ptr<recob::Track> track = fTracks.at(iTrack);
-
+    fTracksToHits[track] = std::vector<art::Ptr<recob::Hit>>();
```

Comment:
Because `fTracksToHits` is an `std::map`, calling `fTracksToHits[track]` automatically creates a default-constructed `std::vector<art::Ptr<recob::Hit>>`.  I therefore suggest:

```suggestion
    fTracksToHits[track];
```

and wherever else you made a similar change.

Reply:
ah that's a really good point! changes made :)